components/TextFields/src/MDCTextField.m

```diff
@@ -0,0 +1,530 @@
+}
+
+- (UIColor *)underlineColor {
+  return _coordinator.underlineColor;
+}
+
+- (void)setUnderlineColor:(UIColor *)underlineColor {
+  _coordinator.underlineColor = underlineColor;
+}
+
+- (CGFloat)underlineHeight {
+  return _coordinator.underlineHeight;
+}
+
+- (void)setUnderlineHeight:(CGFloat)underlineHeight {
+  _coordinator.underlineHeight = underlineHeight;
+}
+
+#pragma mark - UITextField Property Overrides
+
+- (NSAttributedString *)attributedPlaceholder {
+  return _coordinator.attributedPlaceholder;
+}
+
+- (void)setAttributedPlaceholder:(NSAttributedString *)attributedPlaceholder {
+  [super setAttributedPlaceholder:attributedPlaceholder];
+  _coordinator.attributedPlaceholder = attributedPlaceholder;
+}
+
+- (void)setFont:(UIFont *)font {
+  [super setFont:font];
+  [_coordinator didSetFont];
+}
+
+- (void)setEnabled:(BOOL)enabled {
+  [super setEnabled:enabled];
+  _coordinator.enabled = enabled;
+}
+
+- (NSString *)placeholder {
+  return self.coordinator.placeholder;
+}
+
+- (void)setPlaceholder:(NSString *)placeholder {
+  [super setPlaceholder:placeholder];
+  [self.coordinator setPlaceholder:placeholder];
+}
+
+- (void)setText:(NSString *)text {
+  [super setText:text];
+  [_coordinator didSetText];
+  [[NSNotificationCenter defaultCenter] postNotificationName:MDCTextFieldTextDidSetTextNotification
+                                                      object:self];
+}
+
+#pragma mark - UITextField Overrides
+
+// This method doesn't have a positioning delegate mirror per se. But it uses the
+// textContainerInsets value the positioning delegate can return to inset this text rect.
+- (CGRect)textRectForBounds:(CGRect)bounds {
+  CGRect textRect = bounds;
+
+  // Standard textRect calculation
+  UIEdgeInsets textContainerInset = [_coordinator textContainerInset];
+  if (self.mdc_effectiveUserInterfaceLayoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    textRect.origin.x += textContainerInset.right;
+  } else {
+    textRect.origin.x += textContainerInset.left;
+  }
+  textRect.size.width -= textContainerInset.left + textContainerInset.right;
+
+  // Adjustments for .leftView, .rightView
+  CGFloat leftViewWidth = CGRectGetWidth([self leftViewRectForBounds:bounds]);
+  CGFloat rightViewWidth = CGRectGetWidth([self rightViewRectForBounds:bounds]);
+  rightViewWidth += MDCTextInputTextRectRightPaddingCorrection;
+
+  if (self.leftView.superview) {
+    textRect.size.width -= leftViewWidth;
+  }
+  if (self.rightView.superview) {
+    textRect.size.width -= rightViewWidth;
+  }
+
+  // Adjustments for RTL and clear button
+  // .leftView and .rightView actually are leading and trailing: their placement is reversed in RTL.
+  CGFloat scale = [UIScreen mainScreen].scale;
+  CGFloat clearButtonWidth = self.clearButtonImage.size.width / scale;
+  clearButtonWidth += MDCTextInputTextRectRightPaddingCorrection;
+
+  if (self.mdc_effectiveUserInterfaceLayoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
```

Comment:
I don't know if it will work in this case, but it may simplify things to calc the textRect as if it was in standard LTR mode.

Then at the very end of the method:
if (self.mdc_effectiveUIDirection == RTL) {
  textRect = MDCRectFlippedForRTL(textRect, CGRectGetWidth(bounds), RTL);
}